### PR TITLE
feat: migrate partner-chains-demo into partner-chains

### PR DESCRIPTION
**Pull request by skylar-simoncelli**

### CODE REVIEW

**Comment by nrdxp** [comment]
Overall a good idea, just some notes.

We discussed yesterday in the leadership sync that `demo` is probably not a good, descriptive name for this. At least Karen was confused by it. Maybe something more descriptive like `local-env` or something that describes what exactly it is and what it is used for.

Also this would probably be better placed under the top-level `dev` directory I created in #69, once that is merged.

**Comment by skylar-simoncelli** [comment]
Thank you @nrdxp, I am good with "local environment"

I have updated the ASCII art and moved the project to `/dev/local-environment/`

File: dev/local-environment/configurations/cardano/cardano-node-2/config-pool2.json [hunk 1]
```diff
@@ -0,0 +1,117 @@
+{
```
Comment (LGLO) [inline dev/local-environment/configurations/cardano/cardano-node-2/config-pool2.json:1]: This is same file as config-pool1.json
IMO should be moved to shared or maybe there is some ports customization missing here?

File: dev/local-environment/configurations/cardano/cardano-node-2/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#! /bin/bash
+
+chmod 600 /keys/*
+
+echo "Waiting for Node 1 to prepare configuration files and set the target start time..."
```
Comment (LGLO) [inline dev/local-environment/configurations/cardano/cardano-node-2/entrypoint.sh:5]: Replace Node 1 with Node 2 in this entrypoint

File: dev/local-environment/configurations/cardano/cardano-node-3/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#! /bin/bash
+
+chmod 600 /keys/*
+
+echo "Waiting for Node 1 to prepare configuration files and set the target start time..."
```
Comment (LGLO) [inline dev/local-environment/configurations/cardano/cardano-node-3/entrypoint.sh:5]: Replace Node 1 with Node 3 in this file

File: dev/local-environment/configurations/cardano/cardano-node-2/keys/pledge [hunk 4]
```diff
@@ -0,0 +1 @@
+1000000000000
```
Comment (LGLO) [inline dev/local-environment/configurations/cardano/cardano-node-2/keys/pledge:1]: Are pledge and port files used anywhere? I can see `cat` for them, but maybe I've missed it. If they are not used, please don't commit them into repo.

File: dev/local-environment/configurations/partner-chains-nodes/partner-chains-node-5/keys/payemtn.short.addr
Comment (LGLO) [inline dev/local-environment/configurations/partner-chains-nodes/partner-chains-node-5/keys/payemtn.short.addr:1]: This file name is wrong

File: dev/local-environment/configurations/kupo/entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+exec /bin/kupo \
+  --node-socket=/node-ipc/node.socket \
+  --node-config=/shared/node-1-config.json \
+  --host=0.0.0.0 \
+  --workdir=/kupo-workdir \
+  --match=* \
+  --since=origin &
+
+  touch /shared/kupo.ready
```
Comment (LGLO) [inline dev/local-environment/configurations/kupo/entrypoint.sh:23]: I would certainly replace this with waiting for an open port instead

File: dev/local-environment/configurations/ogmios/entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+done
+
+echo "Cardano chain ready. Starting Ogmios..."
+
+exec /bin/ogmios \
+  --host=0.0.0.0 \
+  --node-config=/shared/node-1-config.json \
+  --node-socket=/node-ipc/node.socket &
+
+  touch /shared/ogmios.ready
```
Comment (LGLO) [inline dev/local-environment/configurations/ogmios/entrypoint.sh:20]: I would certainly replace this with waiting for an open port instead

File: dev/local-environment/configurations/pc-contracts-cli/entrypoint.sh [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+
+while true; do
+    if nc -z kupo $KUPO_PORT && nc -z ogmios $OGMIOS_PORT; then
+        break
+    else
+        sleep 10
+    fi
+done
+
+echo "Kupo and Ogmios have started. Waiting 10 seconds for them to stabilise, and then will begin configuration..."
```
Comment (LGLO) [inline dev/local-environment/configurations/pc-contracts-cli/entrypoint.sh:83]: Ogmios is stateless, so it doesn't need time to sync. Kupo promises a header that tells you slot of the last block it knows, but I have not seen it in the output, so perhaps we have to settle on this.
Writing this comment to this env developers to be aware.

**Review by LGLO — APPROVED** [review]
I did my best. I hope it will work. Please do test locally before merging. I think there is still some room for improvement, but I understand this code is moved and I don't want to force one tasked with moving it with improving it.

**Comment by skylar-simoncelli** [comment]
> I did my best. I hope it will work. Please do test locally before merging. I think there is still some room for improvement, but I understand this code is moved and I don't want to force one tasked with moving it with improving it.

Thank you so much Lech, fantastic feedback and valuable suggestions. I will try to work in as much as I can before merging

**Pull request merged**